test/jdk/java/awt/print/PrinterJob/PrintAllFonts.java

```diff
@@ -21,51 +21,53 @@
  * questions.
  */
 
-/**
- *
- * test
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.GraphicsEnvironment;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+
+/*
+ * @test
  * @bug 4884389 7183516
  * @key printer
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary Font specified with face name loses style on printing
  * @run main/manual PrintAllFonts
  */
-
-import java.awt.*;
-import java.awt.print.*;
-import java.awt.GraphicsEnvironment;
-
 public class PrintAllFonts implements Printable {
-
-    static Font[] allFonts;
-    int fontNum = 0;
-    int startNum = 0;
-    int lineHeight = 18;
-    boolean done = false;
-    int thisPage = 0;
-
+    private static final int LINE_HEIGHT = 18;
+    private static final int FONT_SIZE = 14;
+    private final Font[] allFonts =
+            GraphicsEnvironment.getLocalGraphicsEnvironment().getAllFonts();
+    private int fontNum = 0;
+    private int startNum = 0;
+    private int thisPage = 0;
+    private static final String INSTRUCTIONS =
+            "This bug is system dependent and is not always reproducible.\n" +
+            "Font names will be printed in two columns, normal and italic.\n" +
+            "A passing test will have all text printed with correct font style.";
 
     public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
 
-        String[] instructions =
-        {
-            "You must have a printer available to perform this test and should use Win 98.",
-            "This bug is system dependent and is not always reproducible.",
-            " ",
-            "A passing test will have all text printed with correct font style.",
-        };
-
-        Sysout.createDialog( );
-        Sysout.printInstructions( instructions );
-
-        GraphicsEnvironment ge =
-            GraphicsEnvironment.getLocalGraphicsEnvironment();
-        allFonts = ge.getAllFonts();
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
```

Comment:
```suggestion
                .instructions(INSTRUCTIONS)
                .testTimeOut(10)
                .rows((int) INSTRUCTIONS.lines().count() + 1)
```
The list of fonts could be long, and the tester may need to go to fetch the printouts from a printer.